86cueubmz/UI/trip-booking-page

abhijith-biju opened the pull request:
## Overview
This pull request introduces addition of trip-booking-page UI.

## Changes Made

### 1. Trip Booking Page
- Added TripBookingPage component which includes trip details accordion and booking details form.
- Implemented responsive booking details form using formik and added validation using yup.
- Added multi-lingual support for the page. 

## UI Reference
![Trip Booking Page](https://github.com/nijin-26/bus-ticket-booking-react/assets/146805263/c9deac99-c215-4ee0-a0de-44f5c696bb24)

abhijith-biju committed:
feat(tripBookingPage): add trip booking page

abhijith-biju committed:
feat(tripBookingPage): add passenger details input fields

abhijith-biju committed:
Merge branch 'develop' of github.com:amaljoeqb/bus-ticket-booking-react into 86cueubmz/UI/Trip-Booking-Page

abhijith-biju committed:
refactor(passengerDetails): add passengerDetails component

abhijith-biju committed:
fix(passengerDetails): bug in gender select

abhijith-biju committed:
feat(tripBookingForm): add formik wrap

abhijith-biju committed:
feat(passengerDetails): add validation

abhijith-biju committed:
fix(passengerDetails): typo in validation message

abhijith-biju committed:
Merge branch 'develop' of github.com:amaljoeqb/bus-ticket-booking-react into 86cueubmz/UI/Trip-Booking-Page

abhijith-biju committed:
feat(passengerDetails): add spanish language support

abhijith-biju committed:
Merge branch 'develop' of github.com:amaljoeqb/bus-ticket-booking-react into 86cueubmz/UI/Trip-Booking-Page

abhijith-biju committed:
feat(passengerDetails): use Formik FastFields
- replaced Field with FastField for performance improvement
- edited trip booking page URL path

abhijith-biju committed:
style(tripBookingPage): add page heading

abhijith-biju committed:
Merge branch 'develop' of github.com:amaljoeqb/bus-ticket-booking-react into 86cueubmz/UI/Trip-Booking-Page

abhijith-biju committed:
feat(tripBookingPage): add trip accordion

abhijith-biju committed:
refactor(tripBooking): folder structure refactor

abhijith-biju committed:
style(passengerDetails): gender field styles

abhijith-biju committed:
Merge branch 'develop' of github.com:amaljoeqb/bus-ticket-booking-react into 86cueubmz/UI/Trip-Booking-Page

abhijith-biju committed:
style(tripBookingPage): accordion default state
- make trip accordion in expanded state by default in trip booking page

abhijith-biju committed:
fix(passengerDetails): bug when changing language

RJR-003 commented on src/pages/TripBookingPage/PassengerDetailsForm/PassengerDetailsForm.tsx:
```diff
@@ -0,0 +1,150 @@
+                                <Box
+                                    key={index}
+                                    component={Paper}
+                                    py="10px"
```
can use rem values instead of px values.

abhijith-biju replied:
updated.

RJR-003 commented on src/pages/TripBookingPage/TripBookingPage.tsx:
```diff
@@ -0,0 +1,26 @@
+export const TripBookingPage = () => {
+    return (
+        <>
+            <Stack direction="row" alignItems="center" mt={4} mb={3} gap="20px">
```
use rem

abhijith-biju replied:
updated.

RJR-003 reviewed (changes requested):
The first heading should be h1 in a page.
please change all px values to rem

abhijith-biju commented:
resolved all comments.

abhijith-biju committed:
fix(passengerDetails): resolve MR comments
- change tags for headings.
- change px values to rem.

abhijith-biju committed:
fix(passengerDetails): bug when changing language
- fix bug where helperText not updated to new language when language
is changed.

abhijith-biju committed:
style(FareDetails): change in UI

akshay-krishna commented on src/components/ui/TripCardAccordion/AccordionDetails/TripCardDetails.tsx:
```diff
@@ -4,7 +4,7 @@ import SeatLayout from '../../../SeatLayout/SeatLayout';
 import { layoutNames, seats } from '../../../SeatLayout/seatConfig';
 import { StyledAlert } from '../../../Alert/Alert.styled';
-import { FareDetails } from '../../../fairDetails/FareDetails';
+import { FareDetails } from '../../../FairDetails/FareDetails';
```
Change the import into FareDetails

abhijith-biju replied:
updated.

akshay-krishna commented on src/config/routes/routes.config.tsx:
```diff
@@ -1,5 +1,10 @@
-import { LandingPage,  TripsListingPage } from '../../pages';
+import {
+    LandingPage,
+    
```
Use prettier in this file

abhijith-biju replied:
updated.

akshay-krishna commented on src/pages/TripBookingPage/PassengerDetailsForm/PassengerDetailsForm.tsx:
```diff
@@ -0,0 +1,165 @@
+            validateOnChange={false}
+            onSubmit={(values, { setSubmitting }) => {
+                setSubmitting(false);
+                alert(JSON.stringify(values, null, 2));
```
Is this alert necessary?

abhijith-biju replied:
Was included for testing purposes.

abhijith-biju committed:
Merge branch 'develop' of github.com:amaljoeqb/bus-ticket-booking-react into 86cueubmz/UI/Trip-Booking-Page

akshay-krishna commented on src/pages/TripBookingPage/PassengerDetailsForm/validation.ts:
```diff
@@ -0,0 +1,27 @@
+            Yup.object().shape({
+                fullName: Yup.string()
+                    .required(t('fullNameRequired'))
+                    .test('valid-fullName', t('fullNameLength'), (value) => {
```
Full name validation fails when we enter blank space

abhijith-biju replied:
corrected.

akshay-krishna reviewed (changes requested):
Add Changelog

abhijith-biju committed:
fix(tripBookingPage): resolved MR comments

abhijith-biju commented:
> Add Changelog
Updated changelog

lekshmi615213 commented on src/pages/TripBookingPage/PassengerDetailsForm/PassengerDetailsForm.tsx:
```diff
@@ -0,0 +1,165 @@
+    useEffect(() => {
+        // Incrementing languageChangeKey to force re-render when language changes
+        setLanguageChangeKey((prevKey) => prevKey + 1);
+        void formikRef.current?.validateForm();
```
is it required in useEffect/ ?

abhijith-biju replied:
I am using [FastField](https://formik.org/docs/api/fastfield) component provided by Formik for better performance when number of form inputs increases.
When using FastField, form labels did not automatically change language when language state has changed.
Similarly for error message language to change, form validation has to run again.

So useEffect was used to trigger the re-render with new language.

lekshmi615213 commented on src/pages/TripBookingPage/PassengerDetailsForm/validation.ts:
```diff
@@ -0,0 +1,28 @@
+                    .test('valid-age', t('AgeRangeCheck'), (value) => {
+                        if (!value) return false;
+                        const age = parseInt(value);
+                        return age >= 1 && age <= 130;
```
130 :)

abhijith-biju committed:
Merge branch 'develop' of github.com:amaljoeqb/bus-ticket-booking-react into 86cueubmz/UI/Trip-Booking-Page
